common/compiler/solidity_test.go

```diff
@@ -39,6 +41,24 @@ func skipWithoutSolc(t *testing.T) {
 	}
 }
 
+func TestExtractSourceVersion(t *testing.T) {
+	tests := []struct {
+		source   string
+		expected []string
+	}{
+		{"pragma solidity ^0.4.24;\n", []string{"^0.4.24"}},
+		{"pragma solidity 0.4.24;\n", []string{"0.4.24"}},
+		{"pragma solidity ^0.4.24;\n", []string{"^0.4.24"}},
+		{"pragma solidity 0.4.24;\npragma solidity 0.4.25;\n", []string{"0.4.24", "0.4.25"}},
+		{"pragma solidity 0.4.24;\n//pragma solidity 0.4.25;\n", []string{"0.4.24"}},
+		{"//pragma solidity 0.4.24;\npragma solidity 0.4.25;\n", []string{"0.4.25"}},
```

Comment:
Missing test cases may be added such as `pragma solidity >=0.6.0 <0.9.0;`

Reply:
I removed the solidity version parsing logic from the previous request.
This latst change just compiles the target contract with the available solc and loads bytecode if it failed.